source/framework/vx_node_api.c

```diff
@@ -476,6 +476,14 @@ VX_API_ENTRY vx_node VX_API_CALL vxAccumulateWeightedImageNode(vx_graph graph, v
                                    dimof(params));
 }
 
+VX_API_ENTRY vx_node VX_API_CALL vxAccumulateWeightedImageNodeX(vx_graph graph, vx_image input, vx_float32 alpha, vx_image accum)
+{
+    vx_scalar salpha = vxCreateScalar(vxGetContext((vx_reference)graph), VX_TYPE_FLOAT32, &alpha);
+    vx_node node = vxAccumulateWeightedImageNode(graph, input, salpha, accum);
```

Comment:
This may work since you modified the vxAccumulateWeightedImageNode function, however, the kernel string will not match if queried.  We may need to change the kernel wrapper further to abstract this delta.

Reply:
ok, agree. How do you want to do it? Shall we duplicate the code more?

Reply:
I think there is a way we can minimize the code duplication:

Move most of tivxAddKernelAccumulateWeighted into a local static function.
This function is called by tivxAddKernelAccumulateWeighted  and tivxAddKernelAccumulateWeightedX.  Only difference in these two funcitons is the kernel string and the kernel enums ... rest will just call the new static function.

This will require there to be a second static variable in this file: 
static vx_kernel vx_accumulate_weighted_x_kernel = NULL;

We can duplicate tivxRemoveKernelAccumulateWeighted since it is small enough.

Something similar is needed on the target side, but we can simply duplicate the add and remove functions since they are small:
tivxAddTargetKernelAccumulateWeighted
tivxRemoveTargetKernelAccumulateWeighted

Finally, in the vx_node_api.c file, we duplicate the existing vxAccumulateWeightedImageNode.

This way, we are minimizing duplication just to the high level APIs which add/remove kernels, and create the kernel.  All the callbacks within host and target side can be used as is.

Does this make sense?

Reply:
After further discussion, it was clarified that the new nodes will use the original kernel names.  Both are the same, only difference is if we use a scalar or integral type to the public API.